application/controllers/admin/ConditionsAction.php

```diff
@@ -260,8 +260,22 @@ public function index($subaction, $iSurveyID = null, $gid = null, $qid = null)
             }
         }
 
+
         // Previous question parsing ==> building cquestions[] and canswers[]
         if ($questionscount > 0) {
+            $qids = [];
```

Comment:
Clearly goes out of the mantis issue here.

Reply:
@Shnoulle 

The goals of our work here are:

- changing `getQuestionAttributes` usages so that a question object is passed rather than a number to avoid querying the questions individually
- changing `getQuestionAttributes` to no longer allow the passing of question ids and to expect question object instead
- wherever possible change the usages of `getQuestionAttributes` to load the question objects via a single query

Your approach here was to load the question objects individually and only changing the format of how we are calling `getQuestionAttributes`. But that's missing the point.

We intended to reduce the number of queries in the first place and your PR only improved upon the method format, but not upon the usage of the method.

As a result, I have implemented the improvement myself, that is, I'm collecting the question ids into an array, that's the `$qids` variable for, **load the question with a single query rather than querying them separately** via

```
$rawQuestions = Question::model()->findAllByPk($qids);
```

and adding the resulting question objects to `$theserows` records, like we agreed last week. Consequently, we pass these question objects to `getQuestionAttributes` at the line of

```
                $attr = QuestionAttribute::model()->getQuestionAttributes($rows['qObject']);
```

instead of the `$attr = QuestionAttribute::model()->getQuestionAttributes(Question::model()->findByPk($rows['qid']));` that you had.

Reply:
> wherever possible change the usages of `getQuestionAttributes` to load the question objects via a single query

here restricted to type, it's really seems complex code (in my opinion) for this task.

Reply:
I add the _wherever possible change the usages of getQuestionAttributes to load the question objects via a single query_ in the mantis issue : it was not clear for me.

Else : here we can avoid the first foreach : something like that : 
```
// In case we need more type
$neededTypes = [QuestionType::QT_1_ARRAY_DUAL];
$criteria = new CdbCriteria();
$criteria->condition = "parent_qid = 0"
$criteria->compare("sid", $sureyid);
$criteria->addInCondition("type", $neededTypes);
$aQuestionOfNeededTypes = Question::model()->findAll($criteria);
```

And maybe after : just set the keys for $aQuestionOfNeededTypes  to qid.